app/models/submission.rb

```diff
@@ -249,13 +250,12 @@ def make_remark_result
                                 markable_type: extra_mark.markable_type)
     end
 
-    original_result.marks.each do |mark|
-      remark.marks.create(result: remark,
-                          created_at: Time.zone.now,
-                          markable_id: mark.markable_id,
-                          mark: mark.mark,
-                          markable_type: mark.markable_type)
+    remark.marks.each_with_index do |mark, index|
```

Comment:
I realize this is going to take a little more than this, as the marks are still not associated with a criterion yet, and updating the markable_type and markable_id attributes isn't working as I am expecting it to